src/components/elements/button/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { StyledButton, StyledButtonWrapper } from './button.styles';
+import type { ButtonProps } from './button.types';
+import { MovingBorder } from '../moving-border';
+
+export const Button = async ({
+  variant = 'primary',
+  small = false,
+  type = 'button',
+  withBorder = false,
+  children,
+  as = 'button',
+  icon: Icon,
+  iconWeight = 'fill',
+  onClick,
+  href,
+}: Component<ButtonProps>) => {
+  const base = () => (
+    <StyledButton as={as} variant={variant} small={small} type={type} onClick={onClick} href={href}>
+      {children}
+      {Icon && <Icon size={16} weight={iconWeight} />}
+    </StyledButton>
```

Comment:
Can we consider the case where the button is an `a` use `next/link` component for internal routes and external the regular `a` tag? Also, please add some basic accessibility like `role="button"`, etc.